json_rpc/jsonmarshal.nim

```diff
@@ -9,8 +9,11 @@ proc `%`*(n: byte{not lit}): JsonNode =
 proc `%`*(n: uint64{not lit}): JsonNode =
   result = newJInt(int(n))
 
-proc `%`*(n: ref int|ref int64): JsonNode =
-  result = newJInt(int(n[]))
+proc `%`*(n: ref int | ref int64): JsonNode =
```

Comment:
Why not go all the way and say `ref SomeInteger`?

Reply:
Good point!